contractcourt/channel_arbitrator.go

```diff
@@ -1734,12 +1734,10 @@ func (c *ChannelArbitrator) prepContractResolutions(
 				}
 
 				resKit.Quit = make(chan struct{})
-				resolver := &htlcSuccessResolver{
-					htlcResolution:  resolution,
-					broadcastHeight: height,
-					payHash:         htlc.RHash,
-					htlcAmt:         htlc.Amt,
-				}
+				resolver := newSuccessResolver(
```

Comment:
why not make `resKit` a param as well?

Reply:
yes, make it definitely better! :+1:

Reply:
this also exposed some dirtyness around the quit channel. added clean up commit for that too